zrml/neo-swaps/src/lib.rs

```diff
@@ -58,18 +61,26 @@ mod pallet {
     };
     use zeitgeist_primitives::{
         constants::{BASE, CENT},
-        math::fixed::{bdiv, bmul},
+        math::{
+            checked_ops_res::{CheckedAddRes, CheckedSubRes},
+            fixed::{BaseProvider, FixedDiv, FixedMul, ZeitgeistBase},
+        },
         traits::{CompleteSetOperationsApi, DeployPoolApi, DistributeFees},
         types::{Asset, MarketStatus, MarketType, ScalarPosition, ScoringRule},
     };
     use zrml_market_commons::MarketCommonsPalletApi;
 
+    pub(crate) const STORAGE_VERSION: StorageVersion = StorageVersion::new(1);
+
     // These should not be config parameters to avoid misconfigurations.
+    pub(crate) const EXIT_FEE: u128 = CENT / 10;
     pub(crate) const MIN_SWAP_FEE: u128 = BASE / 1_000; // 0.1%.
-    pub(crate) const STORAGE_VERSION: StorageVersion = StorageVersion::new(0);
     pub(crate) const MAX_SPOT_PRICE: u128 = BASE - CENT / 2;
     pub(crate) const MIN_SPOT_PRICE: u128 = CENT / 2;
     pub(crate) const MIN_LIQUIDITY: u128 = BASE;
+    /// The minimum percentage each new LP position must increase the liquidity by, represented as
+    /// fractional (0.03 represents 3%).
+    pub(crate) const MIN_RELATIVE_LP_POSITION_VALUE: u128 = 3 * CENT; // 3%
```

Comment:
First of all, it prevents a griefing vector, namely spamming the tree with small liquidity amounts and preventing said whale from joining the pool.

That being said, it does introduce the "issue" that small LPs can't join the pool if there's already a whale in the pool. However, let's put this into perspective: The condition that a new LP add at least 3% liquidity essentially means that adding liquidity is only allowed if it's actually worth it. If the amount contributed by an LP is so small that we have to reject it, maybe we didn't need it to begin with.

To give an example, if a whale joins with $100,000 worth of liquidity, then we reject anyone else who isn't willing to add at least $3,000 more. I think that's fair.

Reply:
I think it's a big draw back that whales have the opportunity to get total control of markets not only by getting significantly more %, but also by locking others out. This might hurt adoption.

Reply:
Update: I thought liquidity has to be increased by `1 + MIN_RELATIVE_LP_POSITION_VALUE`, which seems not to be the case. This obviously would be an issue.  Since it only has to be increased by `MIN_RELATIVE_LP_POSITION_VALUE` at least, my concerns are lifted.